test/livebook_web/live/session_live_test.exs

```diff
@@ -1171,6 +1171,199 @@ defmodule LivebookWeb.SessionLiveTest do
              |> has_element?()
     end
 
+    test "configuring k8s runtime", %{conn: conn, session: session} do
+      {:ok, view, _} = live(conn, ~p"/sessions/#{session.id}/settings/runtime")
+
+      Session.subscribe(session.id)
+
+      Req.Test.stub(Livebook.Runtime.K8s, Livebook.ClusterStub)
+      Req.Test.stub(Livebook.Runtime.K8s.NoPermissionCluster, Livebook.NoPermissionClusterStub)
+
+      view
+      |> element("#runtime-settings-modal button", "Kubernetes Pod")
+      |> render_click()
+
+      # Check context switcher and switch to context with no permission
+
+      view
+      |> element(~s{form[phx-change="set_context"]})
+      |> render_change(%{context: "no-permission"})
+
+      rendered = render_async(view)
+
+      assert rendered =~ "Authenticated user has no permission to"
+      refute rendered =~ "You can fully customize"
+
+      # Test cluster with full access
+
+      view
+      |> element(~s{form[phx-change="set_context"]})
+      |> render_change(%{context: "default"})
+
+      render_async(view)
+
+      view
+      |> element(~s{form[phx-change="set_namespace"]})
+      |> render_change(%{namespace: "default"})
+
+      assert view
+             |> element(~s{select[name="home_pvc"] option[value="foo-pvc"]})
+             |> has_element?()
+
+      assert view
+             |> element(~s{select[name="home_pvc"] option[value="new-pvc"]})
+             |> has_element?()
+
+      assert render_async(view) =~ "You can fully customize"
+
+      # Create new PVC
+
+      view
+      |> element(~s{button[phx-click="new_pvc"]})
+      |> render_click()
+
+      assert view
+             |> element(~s{form[phx-submit="create_pvc"] input[name="pvc[name]"]})
+             |> has_element?()
+
+      assert view
+             |> element(
+               ~s{form[phx-submit="create_pvc"] input[name="pvc[size_gb]"][type="number"]}
+             )
+             |> has_element?()
+
+      assert view
+             |> element(
+               ~s{form[phx-submit="create_pvc"] select[name="pvc[access_mode]"] option[value="ReadWriteOnce"][selected]}
+             )
+             |> has_element?()
+
+      assert view
+             |> element(
+               ~s{form[phx-submit="create_pvc"] select[name="pvc[storage_class]"] option[value="first-storage-class"]}
+             )
+             |> has_element?()
+
+      assert view
+             |> element(
+               ~s{form[phx-submit="create_pvc"] select[name="pvc[storage_class]"] option[value="second-storage-class"]}
+             )
+             |> has_element?()
+
+      assert view
+             |> element(~s{form[phx-submit="create_pvc"] button[type="submit"][disabled]})
+             |> has_element?()
```

Comment:
Instead of checking for each element, it should be enough to check if the form is there, no? But then, we don't need to do that either, because `render_change` below would fail if the form was missing.

Reply:
true in some cases. I still want to test that the storage_class options are there as they are loaded dynamically. And I'd like to test the logic of disabling the submit button. i.e. disabled before and enabled after validation. wdyt?

Reply:
I'll also add a cancel button intermezzo